internal/weldr/api.go

```diff
@@ -1749,16 +1749,20 @@ func (api *API) composeImageHandler(writer http.ResponseWriter, request *http.Re
 
 	imageBuild := compose.ImageBuilds[0]
 	imageType, _ := imageBuild.ImageType.ToCompatString()
-	imageName, imageMime, err := api.distro.FilenameFromType(imageType)
-
+	archStruct, err := api.distro.GetArch(api.arch.Name())
+	common.PanicOnError(err) // This must not fail because api.arch is predefine, not a user input
```

Comment:
You can just use `api.arch` directly. These sort of "errors that can never happen" are usually a sign that something is wrong.